ff/handlers/nonbonded.py

```diff
@@ -161,6 +162,89 @@ def parameterize_ligand(params, param_idxs):
     return params[param_idxs]
 
 
+def compute_or_load_am1_charges(mol):
+    """Unless already cached in mol's "AM1Cache" property, use OpenEye to compute AM1 partial charges."""
+
+    # check for cache
+    cache_key = "AM1Cache"
+    if not mol.HasProp(cache_key):
+        # The charges returned by OEQuacPac is not deterministic across OS platforms. It is known
+        # to be an issue that the atom ordering modifies the return values as well. A follow up
+        # with OpenEye is in order
+        # https://github.com/openforcefield/openff-toolkit/issues/983
+        am1_charges = list(oe_assign_charges(mol, "AM1"))
+
+        mol.SetProp(cache_key, base64.b64encode(pickle.dumps(am1_charges)))
+
+    else:
+        am1_charges = pickle.loads(base64.b64decode(mol.GetProp(cache_key)))
+
+    return np.array(am1_charges)
+
+
+def bond_smirks_matches(mol, smirks_list):
+    """Return an array of ordered bonds and an array of their assigned types
+
+    Notes
+    -----
+    * Uses OpenEye for substructure searches
+    * Order within smirks_list matters
+        "First match wins."
+        For example, if bond (a,b) can be matched by smirks_list[2], smirks_list[5], ..., assign type 2
+    * Order within each smirks pattern matters
+        For example, "[#6:1]~[#1:2]" and "[#1:1]~[#6:2]" will match atom pairs in the opposite order
+    """
+    oemol = convert_to_oe(mol)
+    AromaticityModel.assign(oemol)
+
+    bond_idxs = []  # [B, 2]
+    type_idxs = []  # [B]
+
+    for type_idx, smirks in enumerate(smirks_list):
+        matches = oe_match_smirks(smirks, oemol)
+
+        for matched_indices in matches:
+            a, b = matched_indices[0], matched_indices[1]
+            forward_matched_bond = [a, b]
+            reverse_matched_bond = [b, a]
+
+            already_assigned = forward_matched_bond in bond_idxs or reverse_matched_bond in bond_idxs
+
+            if not already_assigned:
+                bond_idxs.append(forward_matched_bond)
```

Comment:
@proteneer : could you double-check if there's anything lost by applying this simplification? 

Previously: https://github.com/proteneer/timemachine/blob/466dde7af309aee942e863403db1d738ef52e2e6/ff/handlers/nonbonded.py#L392-L401

Reply:
This looks correct, but I didn't write this originally (it was ported from Simon's BCC code originally). But semantically it seems to simply be: "matched either fwd or rev" and "not already assigned". At some point we may want to be consistent with the rest of OFF and have it "last match wins", though I'm not sure if simply reversing the SMIRKS patterns would accomplish this...